client/src/features/action/needs/NeedFormGoods.tsx

```diff
@@ -53,10 +65,16 @@ const initialFormData: ShareANeedData = {
   deliveryMethod: '',
 };
 
-function NeedForm() {
+function NeedForm(): JSX.Element {
   const [formData, setFormData] = React.useState<ShareANeedData>(initialFormData);
+  const [categories, setCategories] = React.useState<Option[]>([]);
   const history = useHistory();
 
+  (async function () {
```

Comment:
wrap this in a useEffect with 0 dependencies
(will only fire once on component mount)
will still need the iife

Reply:
(otherwise i think this would run on every rerender)